src/blueprint.ts

```diff
@@ -179,6 +193,11 @@ export default class Blueprint extends Command {
       });
       let destRelativepath = filenameTemplate(data);
       let destAbsolutepath = path.join(dest, destRelativepath);
+      let basename = path.basename(destRelativepath);
+      
+      if (renamedFiles[basename]) {
+          destRelativepath = destRelativepath.replace(basename, renamedFiles[basename]);
```

Comment:
Is there any danger here of basename matching part of the filepath that's actually not the basename?

Perhaps something like:

```js
destRelativepath = path.join(path.dirname(destRelativepath), renamedFiles[basename]);
```

Would avoid any chance of that?